### PR TITLE
Improve docs about junit_family warning message

### PR DESCRIPTION
**Pull request by nicoddemus**
From discussion in #6178

cc @bittner

### CODE REVIEW

File: doc/en/deprecations.rst [hunk 1]
```diff
@@ -80,22 +80,35 @@ Note that ``from_parent`` should only be called with keyword arguments for the p
...
+that manipulate this type of file (for example, Jenkins, Azure Pipelines, etc.).
 
-In order to smooth the transition, pytest will issue a warning in case the ``--junitxml`` option
-is given in the command line but ``junit_family`` is not explicitly configured in ``pytest.ini``::
+Users are recommended to try the new ``xunit2`` format and see if their tooling that consumes the JUnit
+XML file supports it.
 
-    PytestDeprecationWarning: The 'junit_family' default value will change to 'xunit2' in pytest 6.0.
-      Add 'junit_family=legacy' to your pytest.ini file to silence this warning and make your suite compatible.
+To use the new format, update your ``pytest.ini``:
```
Comment (bittner) [inline doc/en/deprecations.rst:90]: That suggests that by not adding `junit_family=xunit2` the modern format will not be used. That's not true IIUC, is it?

Comment (nicoddemus) [reply doc/en/deprecations.rst:90]: It is True as per pytest <6.0, only in 6.0 the default will be `xunit2`.

Comment (bittner) [reply doc/en/deprecations.rst:90]: Maybe mentioning this fact may be helpful, e.g.
```suggestion
To use the new format in pytest <6.0, update your ``pytest.ini``:
```

Comment (nicoddemus) [reply doc/en/deprecations.rst]: Good point, thanks!

**Comment by nicoddemus** [comment]
@bittner thanks for the review!

**Comment by jaraco** [comment]
This looks good to me. Perhaps the docs should keep a list of environments that are known to work with the new format rather than to ask each user to perform the experiment. I’ve already validated Azure Pipelines.

**Comment by nicoddemus** [comment]
> Perhaps the docs should keep a list of environments that are known to work with the new format

I will add Jenkins and Azure then, then users are welcome to contribute to update the list.

**Comment by nicoddemus** [comment]
Thanks @jaraco and @bittner for the review.

**Pull request merged**